### PR TITLE
Chore: validate build variables

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This should make the pipelines (wherever they end up) easier to manage, since we'll get warnings as early as possible.

### CODE REVIEW

**Review by raisedadead — CHANGES_REQUESTED** [review]
While you are cleaning up servicebot, lets remove everything about it? A partial remove kind of leaves it dangling.

**Comment by ojeytonwilliams** [comment]
@raisedadead if you're happy I didn't rip out too much, let me know and I'll clean up the commits.

**Review by raisedadead — DISMISSED** [review]
Looks good, if you can cleanup the commits should be ready to go in.

File: tools/scripts/build/ensure-env.js [hunk 1]
```diff
@@ -15,6 +15,39 @@ const globalConfigPath = path.resolve(__dirname, '../../../config');
...
+  expectedVariables.sort();
+  variables.sort();
+  if (expectedVariables.length !== variables.length) {
+    throw Error(`Env. variable validation failed. Expected
+    ${variables} to match
+    ${expectedVariables}
+    `);
+  }
+
+  for (const key of variables) {
```
Comment (ojeytonwilliams) [inline tools/scripts/build/ensure-env.js:44]:
```suggestion
  for (const key of expectedVariables) {
```
I'll make this change this while I rebase.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**